gaeun/app.js

```diff
@@ -60,11 +60,173 @@ app.post("/users", async (req, res) => {
+
+  const post = await myDataSource.query(
+    `INSERT INTO posts(
+      title,
+      post_image,
+      content,
+      user_id
+    ) VALUES (?, ?, ?, ?);
+    `,
+    [title, postImage, content, userId]
+  );
+  res.status(201).json({ message: "postCreated" });
+});
+
+////////////////////////////////
+// Assignment4 - 전체 게시글 조회 //
+////////////////////////////////
+
+app.get("/posts", async (req, res) => {
+  await myDataSource.query(
+    `SELECT
+      u.id AS userId,
+      u.profile_image AS userProfileImage,
+      p.id AS postingId,
+      p.post_image AS postingImageUrl,
+      p.content AS postingContent
+    FROM users u
+    INNER JOIN posts p ON u.id = p.user_id;
+    `,
+    (err, rows) => {
+      res.status(200).json({ data: rows });
+    }
+  );
+});
+
+////////////////////////////////
+// Assignment5 - 유저 게시글 조회 //
+////////////////////////////////
+
+app.get("/post/userId/:userId", async (req, res) => {
+  const { userId } = req.params;
+
+  const [postsList] = await myDataSource.query(
+    `SELECT
+      u.id AS userId,
+      u.profile_image AS userProfileImage,
+      JSON_ARRAYAGG(
+        JSON_OBJECT(
+          'postingId', p.id,
+          'postingImageUrl', p.post_image,
+          'postingContent', p.content
+        )
+      ) AS postings
+    FROM posts p
+    INNER JOIN users u ON u.id = p.user_id
+    WHERE u.id = ?
+    GROUP BY u.id;
+    `,
+    [userId]
+  );
+
+  res.status(200).json({ data: postsList });
+});
+
+/////////////////////////////
+// Assignment6 - 게시글 수정 //
+////////////////////////////
+
+app.patch("/post/edit/:postId", async (req, res) => {
+  const { postId } = req.params;
+  const { content } = req.body;
+
+  // 게시글 수정
+  await myDataSource.manager.query(
+    `UPDATE
+      posts
+    SET
+      content = ?
+    WHERE posts.id = ?`,
+    [content, postId]
+  );
+
+  // 반환
+  const [post] = await myDataSource.query(
+    `SELECT
+      u.id AS userId,
+      u.name AS userName,
+      p.id AS postignId,
+      p.title AS postingTitle,
+      p.content AS postingContent
+    FROM users u
+    INNER JOIN posts p ON u.id = p.user_id
+    WHERE p.id = ?`,
+    [postId]
+  );
+
+  res.status(200).json({ data: post });
+});
+
+/////////////////////////////
+// Assignment7 - 게시글 삭제 //
+////////////////////////////
+
+app.delete("/post/delete/:postId", async (req, res) => {
+  const { postId } = req.params;
+
+  await myDataSource.query(
+    `DELETE 
+    FROM posts 
+    WHERE posts.id = ?`,
+    [postId]
+  );
+  res.status(200).json({ message: "postingDeleted" });
+});
+
+//////////////////////////////
+// Assignment8 - 좋아요 누르기 //
+/////////////////////////////
+
+app.post("/like", async (req, res) => {
+  const { userId, postId } = req.body;
+
+  const [likesRecord] = await myDataSource.query(
+    `SELECT
+      likes.id
+    FROM likes
+    WHERE likes.user_id = ?
+    AND likes.post_id = ?`,
+    [userId, postId]
+  );
+
+  if (likesRecord.length === 0) {
```

Comment:
```suggestion
  if (!likesRecord) {
```
오직 truth or false만 판단하는 것은 이정도로 충분할 것 같습니다!